### PR TITLE
docs: add tips and notes for SAML configuration and troubleshooting.

### PR DESCRIPTION
**Pull request by FlipperPA**
Follows: https://github.com/pennersr/django-allauth/pull/4014

I hope these tips and notes help someone in the future. SAML is very particular and secure (which is good!), but these tips for a new project would have helped me immensely. Thanks for your efforts!

### CODE REVIEW

File: docs/socialaccount/providers/saml.rst [hunk 1]
```diff
@@ -165,3 +165,27 @@ The SAML provider has the following endpoints:
 - ``/accounts/saml/<organization_slug>/sls/``: Single Logout Service URL.
 
 - ``/accounts/saml/<organization_slug>/metadata/``: Metadata URL.
+
+Tips and Notes on SAML with Django:
```
Comment (pennersr) [inline docs/socialaccount/providers/saml.rst:169]:
```suggestion
Guidelines
**********
```

(make it a heading)

File: docs/socialaccount/providers/saml.rst [hunk 2]
```diff
@@ -165,3 +165,27 @@ The SAML provider has the following endpoints:
...
+
+- Most SAML IdPs require TLS (nee SSL) to be used, making testing with
```
Comment (pennersr) [inline docs/socialaccount/providers/saml.rst:171]: What do you mean by "nee SSL" ?

File: docs/socialaccount/providers/saml.rst [hunk 3]
```diff
@@ -165,3 +165,27 @@ The SAML provider has the following endpoints:
...
+  `runserver` challenging. Make sure to configure Django to use HTTPS.
```
Comment (pennersr) [inline docs/socialaccount/providers/saml.rst:172]: Overall remark: RST needs double backticks -- can you preview this section in an RST viewer and adjust accordingly?

**Review by pennersr — CHANGES_REQUESTED** [review]

File: docs/socialaccount/providers/saml.rst (hunk 2 above)
Comment (FlipperPA) [reply docs/socialaccount/providers/saml.rst:172]: `née` is an adjective meaning "formerly known as", as I've found not everyone knows the TLS acronym. Perhaps I'll just change it to "formerly SSL" to clarify.

Comment (banagale) [reply docs/socialaccount/providers/saml.rst]: "aka" is perhaps a more commonly used western English alternative to née.

**Pull request merged**